Brokerages/Tradier/TradierBrokerage.DataQueueHandler.cs

```diff
@@ -96,206 +92,170 @@ public void Unsubscribe(SubscriptionDataConfig dataConfig)
-                CloseStream();
-                _refreshDelay.Stop();
-            };
-            _refreshDelay.Start();
-        }
+                if (!symbol.IsCanonical())
+                {
+                    var ticker = _symbolMapper.GetBrokerageSymbol(symbol);
+                    if (!_subscribedTickers.ContainsKey(ticker))
+                    {
+                        _subscribedTickers.TryAdd(ticker, symbol);
+                        symbolsAdded = true;
+                    }
+                }
+            }
 
-        /// <summary>
-        /// Get a string list of tickers from the symbol dictionary
-        /// </summary>
-        /// <returns>List of string tickers</returns>
-        private List<string> GetTickers()
-        {
-            var values = Subscriptions.Select(x => x.Value).ToList();
-            Log.Trace("TradierBrokerage.DataQueueHandler.GetTickers(): " + string.Join(",", values));
-            return values;
+            if (symbolsAdded)
+            {
+                SendSubscribeMessage(_subscribedTickers.Keys.ToList());
+            }
+
+            return true;
         }
 
-        /// <summary>
-        /// Create a tick from the tradier stream data:
-        /// </summary>
-        /// <param name="tsd">Tradier stream data obejct</param>
-        /// <returns>LEAN Tick object</returns>
-        private Tick CreateTick(TradierStreamData tsd)
+        private bool Unsubscribe(IEnumerable<Symbol> symbols, TickType tickType)
         {
-            var symbol = Subscriptions.FirstOrDefault(x => x.Value == tsd.Symbol);
-
-            // Not subscribed to this symbol.
-            if (symbol == null) return null;
+            var symbolsRemoved = false;
 
-            // Occassionally Tradier sends trades with 0 volume?
-            if (tsd.TradeSize == 0) return null;
+            foreach (var symbol in symbols)
+            {
+                if (!symbol.IsCanonical())
+                {
+                    var ticker = _symbolMapper.GetBrokerageSymbol(symbol);
+                    if (_subscribedTickers.ContainsKey(ticker))
+                    {
+                        Symbol removedSymbol;
+                        _subscribedTickers.TryRemove(ticker, out removedSymbol);
+                        symbolsRemoved = true;
+                    }
+                }
+            }
 
-            // Tradier trades are US NY time only. Convert local server time to NY Time:
-            var unix = tsd.UnixDate.ConvertInvariant<long>() / 1000;
-            var utc = Time.UnixTimeStampToDateTime(unix);
+            if (symbolsRemoved)
+            {
+                var tickers = _subscribedTickers.Keys.ToList();
 
-            // Occassionally Tradier sends old ticks every 20sec-ish if no trading?
-            if (DateTime.UtcNow - utc > TimeSpan.FromSeconds(10)) return null;
+                // Tradier expects at least one symbol
+                if (tickers.Count > 0)
```

Comment:
Could we send some invalid symbol maybe if `tickers == 0`? So that we do not continue getting all the data